### PR TITLE
New privacy metric shapr

### PR DESCRIPTION
**Pull request by abigailgold**
# Description

New privacy metric based on Shapley values, is more scalable than existing PDTP metric.

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [X] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

New tests for new metric.

**Test Configuration**:
- OS: MacOS 12.6
- Python version: 3.8
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: notebooks/privacy_metric.ipynb [hunk 1]
```diff
@@ -0,0 +1,374 @@
+{
```
Comment (beat-buesser) [inline notebooks/privacy_metric.ipynb:1]: Could you please add a link and description for this notebook to `notebooks/README.md`?

File: notebooks/privacy_metric.ipynb [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+ "cells": [
+  {
```
Comment (beat-buesser) [inline notebooks/privacy_metric.ipynb:3]: What do you think about adding labels to the all the plot axes and add comments to guide the users through the notebook?

File: art/metrics/privacy/membership_leakage.py [hunk 3]
```diff
@@ -109,19 +124,113 @@ def PDTP(  # pylint: disable=C0103
...
     avg_per_sample = np.array([sum(val) / len(val) for val in per_sample])
     worse_per_sample = np.max(per_sample, axis=1)
     std_dev_per_sample = np.std(per_sample, axis=1)
 
     # return avg+worse leakage + standard deviation per sample
     return avg_per_sample, worse_per_sample, std_dev_per_sample
+
+
+def SHAPr(  # pylint: disable=C0103
+    target_estimator: "Classifier",
```
Comment (beat-buesser) [inline art/metrics/privacy/membership_leakage.py:151]: Can we use here `CLASSIFIER_TYPE` from `art.utils`?
```suggestion
    target_estimator: CLASSIFIER_TYPE,
```

**Review by beat-buesser — CHANGES_REQUESTED** [review]
Hi @abigailgold Thank you very much for your PR! I have added a few review comments, what do you think?

**Review by beat-buesser — APPROVED** [review]
